src/lib/semantic-versioning.ts

```diff
@@ -0,0 +1,39 @@
+import { isArray } from "lodash"
+
+export type SemanticVersionNumber = {
+  major: number
+  minor: number
+  patch: number
+}
```

Comment:
There was some talk of specifying the version string more simply e.g. `8.59.0` as we do elsewhere.

But since I want to work with the semver components I want this to be more strongly typed.

I also considered using a Typescript [template literal type](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html) — but I got spooked by its apparent lack of accurate enforcement.

For example, it does not complain about this malformed semver string,  which could result in a consequential parsing error:

<img width=400 src="https://github.com/user-attachments/assets/9b0b36f2-ec33-4c5b-b741-d355fd1c1b58" alt="TS why?" />

Reply:
I kinda like the more structured approach as you've done in the PR. While it might be a tad more verbose, it's clearly structured and typing the right realistic abstraction - the Eigen version.